library.rb

```diff
@@ -51,102 +26,54 @@ def loaded?
   def not_loaded
     puts 'You should load library data using get_data(file) method before
     using any other methods'
+    exit
   end
 
   def no_orders
-    puts "There is no orders in the library yet. You may create them using
+    puts "There are no orders in the library yet. You may create them using
     generate_orders(orders_amount) method"
+    exit
   end
 
-  # def get_data(file)
-  #   puts "Reading library data from file #{file}"
-  # end
+  def get_data(file)
+    begin
+      library = YAML.load_file(file)
+    rescue
+      puts "Couldn't open file #{file}"
+      return
+    end
+    @books = library.books
+    @authors = library.authors
+    @readers = library.readers
+    @orders = library.orders
+  end
 
   def save_data(file)
     puts "Saving library data to file #{file}  ..."
     f = File.new(file, 'w')
     f.write(to_yaml)
     f.close
+  rescue => exception
+    puts "Couldn't save data to file. Exception: #{exception}"
   end
 
-  def generate_orders(orders_amount)
-    if loaded?
-      puts "Generating #{orders_amount} random orders..."
-      @orders = []
-      orders_amount.times do
-        book_index = rand(0...@books.count)
-        reader_index = rand(0...@readers.count)
-        date = Date.today - rand(1..30)
-        @orders << Order.new(@books[book_index].title, @readers[reader_index].name, date)
-      end
-    else
-      not_loaded
-    end
-  end
-
-  def top(objects, top_amount = 1)
-    if objects
-      obj_hash = Hash.new(0)
-      entity = objects[0].class.to_s.downcase
-      if @orders[0].respond_to?(entity)
-        @orders.each { |order| obj_hash[order.public_send(entity)] += 1 }
-        maxs = obj_hash.values.uniq.max(top_amount)
-        top = obj_hash.select { |_, v| maxs.include?(v) }
-        top.sort_by { |_, v| -v }.to_h
-      else
-        puts "There is no such method #{entity} for Order instance"
-      end
+  def top(entity, top_amount = 1)
+    return not_loaded unless loaded?
+    return no_orders unless @orders
```

Comment:
these two lines can be moved to separate method too (they are used in 2 methods)